docs/reference/c++_support.rst

```diff
@@ -0,0 +1,165 @@
+.. meta::
+  :description: This chapter describes the C++ support of the HIP ecosystem
+  ROCm software.
+  :keywords: AMD, ROCm, HIP, C++
+
+*******************************************************************************
+C++ language support
+*******************************************************************************
+
+.. _language_introduction:
+Introduction
+===============================================================================
+
+The main way to harness the power of the ROCm platform is through the usage C++ and HIP
+code. This code is then compiled with a ``clang`` or ``clang++`` compiler. The official
+versions support the HIP platform, but for the most up-to-date feature set use the
+``amdclang`` or ``amdclang++`` installed with your ROCm installation.
+
+The source code will be processed based on the ``C++03``, ``C++11``, ``C++14``, ``C++17``
+or ``C++20`` standards, but supports HIP specific extensions and is subject to certain
+restrictions. The largest restriction is the lack of support of the standard library.
+This is mostly because the SIMD nature of the HIP device makes most of the standard
+library implementations not performant or useful. The important operations are
+implemented in HIP specific libraries like rocPRIM, rocThrust and hipCUB.
```

Comment:
The source code is compiled according to the ``C++03``, ``C++11``, ``C++14``, ``C++17``, and ``C++20`` standards, along with HIP-specific extensions, but is subject to restrictions. The key restriction is the lack of standard library support. This is due to the Single Instruction/Multiple Data (SIMD) nature of the HIP device, which makes most of the standard library implementations not performant or useful. However, essential operations are implemented in HIP-specific libraries such as rocPRIM, rocThrust, and hipCUB.

Reply:
-- Add hyperlinks to the libarries.
-- Do you mean "standard library implementations" (i.e GNU C++) or "standard library features" (i.e iostream)?

Reply:
It is both. Some standard library features are not used, because in a GPU context it makes no sense and some are not used, because the implementation is optimized for CPU and wouldn't be efficient.

Reply:
in conclusion: I think implementation is the proper word to use there

Reply:
Fair enough, let's use implementation.